videoanalyser/app/src/main/java/com/example/videoanalyser/MainActivity.java

```diff
@@ -81,14 +82,12 @@ public void onClick(View view) {
                     e.getCause();
                 }
 
-                rotationOfVideo = mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_ROTATION);
-                List<Bitmap> frameList = mediaMetadataRetriever.getFramesAtIndex(100, 100);
+                totalFrames = Integer.valueOf(mediaMetadataRetriever.extractMetadata(METADATA_KEY_VIDEO_FRAME_COUNT));
+                List<Bitmap> frameList = mediaMetadataRetriever.getFramesAtIndex(0, Math.max(0, totalFrames-safeFrames));
```

Comment:
`safeFrames` are required here because while recording a video at high fps, there are cases when the last frames are not taken correctly.
Eg: Consider a video of time 4.606 seconds taken at 120fps, there will be 553 frames as suggested in metadata.
But when we do the math, we find that there should be 552.72 frames and the last frames is not correctly sotred.

To account this, `safeFrames` value trims some frames from the end.

Reply:
Thanks for the explanation. From your explanation, it sounds like safeFrames = 1 should work (unless a bunch of frames are stored together due to compression?) but I'm not too worried about this since I understand we're trying to do the analysis without storing the video at all.

Reply:
Yeah, I also thought `safeFrames = 1` should be enough but for some `safeFrames = 2` was required else I was getting memory leaks. So just to be on safe side I added it as `10` for now.
Thinking now, I realise that this could be the reason.
>(unless a bunch of frames are stored together due to compression?)